assets/src/edit-story/components/canvas/carousel/index.js

```diff
@@ -93,6 +133,9 @@ function Carousel() {
   const listRef = useRef(null);
   const pageRefs = useRef([]);
 
+  const [carouselSize, setCarouselSize] = useState({});
+  const isCompact = carouselSize.height < COMPACT_CAROUSEL_BREAKPOINT;
+
   const openModal = useCallback(() => setIsGridViewOpen(true), [
     setIsGridViewOpen,
```

Comment:
This is not needed and the eslint rule would not enforce it. It actually surprises me, that it's not reported as an error, though it's technically a non-issue, as this function is guaranteed to be unchanged also by reference. But this can and should be changed to simply:

```js
    const openModal = useCallback(() => setIsGridViewOpen(true), []);
```

Reply:
In this case, do we use `useCallback` to avoid re-renders of:
```jsx
<StyledGridViewButton
            width="24"
            height="24"
            onClick={openModal}
            aria-label={__('Grid View', 'web-stories')}
          />
```
?
so for example if you scroll the list (and update the state with `setScrollPercentage`) there is no need to re-render this button, right?